src/frontend/src/icp/components/core/IcLoaderWallets.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+	import { onDestroy } from 'svelte';
+	import { initIcrcWalletWorker } from '$icp/services/worker.icrc-wallet.services';
+	import { initIcpWalletWorker } from '$icp/services/worker.icp-wallet.services';
+	import type { WalletWorker } from '$icp/types/ic-listener';
+	import { enabledIcNetworkTokens } from '$lib/derived/network-tokens.derived';
+	import type { TokenId } from '$lib/types/token';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let workers: Map<TokenId, WalletWorker> = new Map<TokenId, WalletWorker>();
+
+	const manageWorkers = async () => {
+		const unusedWorkers: TokenId[] = Array.from(workers.keys()).filter(
+			(tokenId) => !$enabledIcNetworkTokens.some(({ id: icTokenId }) => icTokenId === tokenId)
+		);
+
+		unusedWorkers.forEach((tokenId) => {
+			const worker = workers.get(tokenId);
+
+			if (nonNullish(worker)) {
+				worker.stop();
+				workers.delete(tokenId);
+			}
+		});
+
+		await Promise.all(
+			$enabledIcNetworkTokens.map(async (token) => {
+				if (!workers.has(token.id)) {
```

Comment:
That's a performance improvement in comparison to existing solution right?

Reply:
Not really sure what you mean: in sense of speed to load?

Reply:
Currently the workers are stop / starter without any logic on "if already mounted or not". Therefore felt like an on purpose performance improvement which is a good thing. Just wanted to ask if it what meant on purpose.

Reply:
ah understood! yes, it was on purpose: i had tested it debugging and that was indeed called like so. So it makes sense to make a sort of pre-check to stop/start